src/couch_mrview/src/couch_mrview.erl

```diff
@@ -203,22 +203,23 @@ validate(DbName, _IsDbPartitioned,  DDoc) ->
                 Msg = ["`", Bad, "` is not a supported reduce function."],
                 throw({invalid_design_doc, Msg});
             ({RedName, RedSrc}) ->
-                couch_query_servers:try_compile(Proc, reduce, RedName, RedSrc)
+                couch_eval:try_compile(Ctx, RedName, RedSrc)
         end, Reds)
     end,
     {ok, #mrst{
         language = Lang,
         views = Views
     }} = couch_mrview_util:ddoc_to_mrst(DbName, DDoc),
 
-    try Views =/= [] andalso couch_query_servers:get_os_process(Lang) of
+    try Views =/= [] andalso couch_eval:acquire_context(Lang) of
```

Comment:
`couch_eval:acqure_context/1` can raise `{invalid_eval_api_mod, Language}` or `{unknown_eval_api_language, Language}. Do we need to handle this on line [224](https://github.com/apache/couchdb/pull/3481/files#diff-637f0ee4524b44455ad1fcbfa99fdbdc757c0a0c7a1bd462f4e9b5f51a548dc9R224)?

Reply:
I don't think so. I'm happy for that error to bubble up.